source/benchmarking/odm-benchmarking.md

```diff
@@ -0,0 +1,393 @@
+- Minimum Server Version: N/A
+
+## Abstract
+
+This document describes a standard benchmarking suite for MongoDB ODMs (Object Document Mappers). Much of this
+document's structure and content is taken from the existing MongoDB driver benchmarking suite for consistency.
+
+## Overview
+
+### Name and purpose
+
+ODM performance will be measured by the MongoDB ODM Performance Benchmark. It will provide both "horizontal" insights
+into how individual ODM performance evolves over time and two types of "vertical" insights: the relative performance of
+different ODMs, and the relative performance of ODMs and their associated language drivers.
+
+We expect substantial performance differences between ODMs based on both their language families (e.g. static vs.
+dynamic or compiled vs. virtual-machine-based) as well as their inherent design (e.g. web frameworks such as Django vs.
+application-agnostic such as Mongoose). Within families of ODMs that use similar design and language families, these
+comparisons could be used to identify potential areas of performance improvement.
+
+### Task Hierarchy
+
+The benchmark suite consists of two groups of small, independent benchmarks. This allows us to better isolate areas
+within ODMs that are faster or slower.
+
+- Flat models -- reading and writing flat models of various sizes, to explore basic operation efficiency
+- Nested models -- reading and writing nested models of various sizes, to explore basic operation efficiency for complex
+    data
+
+The suite is intentionally kept small for several reasons:
+
+- ODM feature sets vary significantly across libraries, limiting the number of benchmarks that can be run across the
+    entire collection of extant ODMs.
+- Several popular MongoDB ODMs are actively maintained by third-parties, such as Mongoose. By limiting the benchmarking
+    suite to a minimal set of representative tests that are easy to implement, we encourage adoption of the suite by
+    these third-party maintainers.
+
+### Measurement
+
+In addition to latency data, all benchmark tasks will be measured in terms of "megabytes/second" (MB/s) of documents
+processed, with higher scores being better. (In this document, "megabyte" refers to the SI decimal unit, i.e. 1,000,000
+bytes.) This makes cross-benchmark comparisons easier.
+
+To avoid various types of measurement skew, tasks will be measured over several iterations. Each iteration will have a
+number of operations performed per iteration that depends on the task being benchmarked. The final score for a task will
+be the median score of the iterations. A range of percentiles will also be recorded for diagnostic analysis.
+
+### Data sets
+
+Data sets will vary by task. In most cases, data sets will be synthetic line-delimited JSON files to be constructed by
+the ODM being benchmarked into the appropriate model. Some tasks will require additional modifications to these
+constructed models, such as adding generated ObjectIds.
```

Comment:
Is this referring to the contents of the `tgz` files in the PR? If so, are these files a single file with one JSON document per line?